cpp/src/centrality/betweenness_centrality_impl.cuh

```diff
@@ -548,27 +548,32 @@ rmm::device_uvector<weight_t> betweenness_centrality(
   std::optional<weight_t> scale_factor{std::nullopt};
 
   if (normalized) {
-    weight_t n = static_cast<weight_t>(graph_view.number_of_vertices());
-    if (!include_endpoints) { n -= weight_t{1}; }
-
-    scale_factor = n * (n - 1);
-  } else if (graph_view.is_symmetric())
+    if (include_endpoints) {
+      if (graph_view.number_of_vertices() >= 2) {
+        scale_factor = static_cast<weight_t>(
+          std::min(static_cast<vertex_t>(num_sources), graph_view.number_of_vertices()) *
+          (graph_view.number_of_vertices() - 1));
+      }
+    } else if (graph_view.number_of_vertices() > 2) {
+      scale_factor = static_cast<weight_t>(
+        std::min(static_cast<vertex_t>(num_sources), graph_view.number_of_vertices() - 1) *
```

Comment:
No need to subtract 1 from `num_sources`? (i.e. `static_cast<vertex_t>(num_sources - 1)`?)

I assume `num_sources == graph_view.number_of_vertices()` for full BC. It looks a bit weird to subtract 1 just from `graph_view.number_of_vertices()`.